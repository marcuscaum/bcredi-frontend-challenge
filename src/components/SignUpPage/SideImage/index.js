import React from 'react';
import PropTypes from 'prop-types';

import {
  SideImage,
  MessageContainer,
  MessageBox,
  MessageContent,
  MessageAuthor,
} from './index.styled';

const SideImageComponent = ({ message, messageAuthor }) => (
  <SideImage>
    <MessageContainer>
      <MessageBox>
        <MessageContent>{message}</MessageContent>
        <MessageAuthor>{messageAuthor}</MessageAuthor>
      </MessageBox>
    </MessageContainer>
  </SideImage>
);

SideImageComponent.propTypes = {
  message: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  messageAuthor: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

export default SideImageComponent;
