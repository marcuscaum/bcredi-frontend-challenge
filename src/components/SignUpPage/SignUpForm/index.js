import React from 'react';

import FormField from 'components/FormField';

const SignUpForm = () => (
  <form>
    <header>
      <h1>Criar meu cadastro</h1>
      <h2>Para acompanhar sua contratação de crédito você utilizará seu e-mail e CPF.</h2>
    </header>
    <FormField label="Nome completo">
      <input type="text" />
    </FormField>
  </form>
);

export default SignUpForm;
