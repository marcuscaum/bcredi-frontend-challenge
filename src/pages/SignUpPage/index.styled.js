import styled from 'styled-components';

export const MainLayout = styled.div`
  display: flex;
  background-color: white;
`;

export const FormContainer = styled.div`
  display: flex;
  width: 320px;
  margin: 40px auto;
  flex-direction: column;
`;

export const Container = styled.div`
  width: 100%;
`;
