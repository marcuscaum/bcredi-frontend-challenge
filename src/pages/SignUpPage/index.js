import React from "react";
import MediaQuery from "react-responsive";

import SideImage from "components/SideImage";
import SignUpForm from "components/SignUpForm";
import NavBar from "components/NavBar";

import { MainLayout, Container, FormContainer } from "./index.styled";

const SignUpPage = () => (
  <MainLayout>
    <MediaQuery minDeviceWidth={700}>
      <SideImage
        messageAuthor={
          <React.Fragment>
            <strong>Camila Bragança,</strong>
            <br />
            Sideral Tecnologia
          </React.Fragment>
        }
        message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
      />
    </MediaQuery>
    <Container>
      <NavBar />
      <FormContainer>
        <SignUpForm />
      </FormContainer>
    </Container>
  </MainLayout>
);

export default SignUpPage;
