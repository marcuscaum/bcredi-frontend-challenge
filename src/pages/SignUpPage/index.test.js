import React from "react";
import { shallow } from "enzyme";

import SignUpPage from "./index";

const setup = props => shallow(<SignUpPage {...props} />);

describe("[Page] SignUpPage", () => {
  it("should render the page", () => {
    const wrapper = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
