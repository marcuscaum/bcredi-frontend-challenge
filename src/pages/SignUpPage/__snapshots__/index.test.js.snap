// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] SignUpPage should render the page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MediaQuery
          minDeviceWidth={700}
          values={Object {}}
>
          <SideImageComponent
                    message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
                    messageAuthor={
                              <Unknown>
                                        <strong>
                                                  Camila Bragança,
                                        </strong>
                                        <br />
                                        Sideral Tecnologia
                              </Unknown>
                    }
          />
</MediaQuery>,
        <styled.div>
          <NavBarComponent />
          <styled.div>
                    <withStateHandlers(SignUpFormComponent) />
          </styled.div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <SideImageComponent
            message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
            messageAuthor={
                        <Unknown>
                                    <strong>
                                                Camila Bragança,
                                    </strong>
                                    <br />
                                    Sideral Tecnologia
                        </Unknown>
            }
/>,
          "minDeviceWidth": 700,
          "values": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido.",
            "messageAuthor": <Unknown>
              <strong>
                            Camila Bragança,
              </strong>
              <br />
              Sideral Tecnologia
</Unknown>,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavBarComponent />,
            <styled.div>
              <withStateHandlers(SignUpFormComponent) />
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withStateHandlers(SignUpFormComponent) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MediaQuery
            minDeviceWidth={700}
            values={Object {}}
>
            <SideImageComponent
                        message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
                        messageAuthor={
                                    <Unknown>
                                                <strong>
                                                            Camila Bragança,
                                                </strong>
                                                <br />
                                                Sideral Tecnologia
                                    </Unknown>
                        }
            />
</MediaQuery>,
          <styled.div>
            <NavBarComponent />
            <styled.div>
                        <withStateHandlers(SignUpFormComponent) />
            </styled.div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <SideImageComponent
              message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
              messageAuthor={
                            <Unknown>
                                          <strong>
                                                        Camila Bragança,
                                          </strong>
                                          <br />
                                          Sideral Tecnologia
                            </Unknown>
              }
/>,
            "minDeviceWidth": 700,
            "values": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "message": "Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido.",
              "messageAuthor": <Unknown>
                <strong>
                                Camila Bragança,
                </strong>
                <br />
                Sideral Tecnologia
</Unknown>,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavBarComponent />,
              <styled.div>
                <withStateHandlers(SignUpFormComponent) />
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <withStateHandlers(SignUpFormComponent) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
